include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -172,24 +175,40 @@ MGTransferPrebuilt<VECTOR>::copy_to_mg (
 {
   reinit_vector(mg_dof_handler, component_to_block_map, dst);
   bool first = true;
+#ifdef DEBUG_OUTPUT
+  std::cout << "copy_to_mg src " << src.l2_norm() << std::endl;
+  MPI_Barrier(MPI_COMM_WORLD);
+#endif
   for (unsigned int level=mg_dof_handler.get_tria().n_global_levels(); level != 0;)
     {
       --level;
       VECTOR &dst_level = dst[level];
 
+#ifdef DEBUG_OUTPUT
+      MPI_Barrier(MPI_COMM_WORLD);
+#endif
+
       typedef std::vector<std::pair<types::global_dof_index, types::global_dof_index> >::const_iterator IT;
       for (IT i= copy_indices[level].begin();
            i != copy_indices[level].end(); ++i)
         dst_level(i->second) = src(i->first);
 
-      // for (IT i= copy_indices_to_me[level].begin();
-      //      i != copy_indices_to_me[level].end(); ++i)
-      //   dst_level(i->second) = src(i->first);
+      for (IT i= copy_indices_global_mine[level].begin();
+           i != copy_indices_global_mine[level].end(); ++i)
+        dst_level(i->second) = src(i->first);
```

Comment:
Ah, I realize I have a question: As I understand it, this code sets the level dofs from the source on a remote processor (since only the global index is mine but not the local one).

This requires that the vector keeps track of non-local indices in a specific way. `parallel::distributed::Vector` does not do that as it throws away ghost data upon `compress(VectorOperation::insert)`. I checked the Trilinos vectors and they seem to be fine **if** each level dof index is **set only once** (or alternatively each remote processor sets the exact same value). I assume this is fine (as multigrid is working) because there is a 1-to-1 relation between global dof indices and level dof indices.

Just to understand this part better: This is exactly equivalent to using the `copy_from_mg_add` version on a vector that was zeroed before, isn't it?